lib/SILOptimizer/Utils/ValueLifetime.cpp

```diff
@@ -281,3 +289,20 @@ void ValueLifetimeAnalysis::dump() const {
   }
   llvm::errs() << '\n';
 }
+
+void swift::endLifetimeAtFrontier(
+    SILValue valueOrStackLoc, const ValueLifetimeAnalysis::Frontier &frontier,
+    SILBuilder &builder) {
```

Comment:
Why pass in a SILBuilder here? Better to pass in a SILBuilderContext and use a locally created SILBuilderWithScope. Also, why are you messing with RegularLocations here?

Reply:
SILBuilder: ok, makes sense.

The frontier instruction can have a ReturnLocation. Therefore I have to construct a RegularLocation.